auth-server/src/controllers/user.ts

```diff
@@ -0,0 +1,77 @@
+import errorHandler from '../model/errors'
+import AuthService from '../services/auth'
+
+/**
+ * POST /login
+ * Sign in using username and password.
+ */
+export const postLogin = async (req, res) => {
+	req.assert('username', 'Username is not valid').notEmpty()
+	req.assert('password', 'Password cannot be blank').notEmpty()
+
+	const errors = req.validationErrors()
+
+	if (errors) {
+		return res.status(400).json({ errors })
+	}
+
+	const username = req.body.username
+	const password = req.body.password
+	try {
+		const authServiceInstance = new AuthService()
+		const { user, token, refreshToken } = await authServiceInstance.Login(username, password)
+		res.cookie('refresh_token', refreshToken, { expires: new Date(Date.now() + 6 * 30 * 24 * 60 * 60 * 1000) })
```

Comment:
We need an httponly cookie. look at the options in the top answers, we need `httponly: true`
Not sure about the `expires` either.
It looks like it should be `maxage` instead.

Reply:
done